servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -408,6 +414,8 @@ boolean isWritable() {
 
         void writeNext(Object msg) {
             assert eventLoop.inEventLoop();
+            assert isWritable() : channel + " Unexpected writeNext: " + msg + " during non-writable state=" +
```

Comment:
is this assertion necessary? wondering if we need to prevent any write-after-close/terminated scenarios as we may want to pass on the object anyways (reference counted, ordered promos completion, ..)

Reply:
Not really, invocation of this method is currently guarded by `isWritable()` check:
```java
    void doWrite(Object msg) {
        // Ignore onNext if the channel is already closed.
        if (promise.isWritable()) {
            long capacityBefore = channel.bytesBeforeUnwritable();
            promise.writeNext(msg);
```

Added this assection just in case the contract changes, to make sure we handle writes correctly as it will likely will also required changes in `AllWritesPromise`.